sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -656,8 +657,11 @@ public void deleteDataset(String projectId, String datasetId)
     }
 
     @VisibleForTesting
-    long insertAll(TableReference ref, List<TableRow> rowList, @Nullable List<String> insertIdList,
-        BackOff backoff, final Sleeper sleeper) throws IOException, InterruptedException {
+    long insertAll(TableReference ref, List<ValueInSingleWindow<TableRow>> rowList,
```

Comment:
ValueInSingleWindow seems out of place here (wrong level of abstraction), and I don't see the insertAll code using anything except the value itself.

I realize that this is being used to let caller code maintain the context from which every errored value originated. Perhaps maintain it by returning a list of indices of failed inserts in `rowList`? Or perhaps still better, return a list of tuples (failed index, error proto)?